include/swift/Sema/IDETypeChecking.h

```diff
@@ -206,6 +206,31 @@ namespace swift {
   /// the decl context.
   ProtocolDecl *resolveProtocolName(DeclContext *dc, StringRef Name);
 
+  /// Reported type of a variable declaration.
+  struct VariableTypeInfo {
+    /// The start of the variable identifier.
+    uint32_t offset;
+
+    /// The length of the variable identifier.
+    uint32_t length;
+
+    /// Whether the variable has an explicit type annotation.
+    bool hasExplicitType;
+
+    /// The start of the printed type in a separate string buffer.
+    uint32_t typeOffset;
+
+    /// The length of the printed type in a separate string buffer.
+    uint32_t typeLength;
+  };
+
+  /// Collect type information for every variable declaration in \c SF;
+  /// All types will be printed to \c OS.
```

Comment:
As you’ll see further below the use of `OS` kept confusing me. I think it would be good to clarify its interplay with `VariableTypeInfo`
```cpp
/// All types will be printed to \c OS and the type offsets of the \c VariableTypeInfos will refer into the string that backs this stream.
```